core.py

```diff
@@ -450,14 +369,19 @@ def compute_heuristic(self, state):
         """Compute the heuristic for a given state in the current problem.
         """
         problem = self.problems[self._problem_idx]
-        state = state | set(self.action_space._all_ground_literals)
+        # Add action literals to state to enable planning
+        state_lits = set(state.literals)
+        action_lits = set(self.action_space.all_ground_literals(state, 
+            valid_only=False))
```

Comment:
Any reason to prefer `valid_only=False`?

Reply:
This is the trick that I rambled about on slack that makes planning with off-the-shelf planners possible, given our distinction between actions and operators. Copying what I said here for posterity:

the reason we introduced action literals early on is because we felt like there’s an important difference between the high-level actions that an agent is hypothetically able to turn into some low-level controls, and the action operators that appear in a PDDL domain file. the high-level actions include only “free parameters” that the agent can choose itself. for example in a grid-world type domain (e.g. sokoban), one action might be (move ?dir) where ?dir is one of up/down/left/right.
in contrast, the action operators that appear in a PDDL file include parameters that are not “free”, but are instead involved in preconditions or effects of the action. for example, instead of (move ?dir), an action operator would be (move ?from ?to ?dir) because ?from and ?to are necessary to check the preconditions and determine the effects of moving in the direction ?dir.
so that’s why we introduced separate high-level actions into the PDDL files. and we put them as preconditions in the action operators, so when an agent takes an action like (move north) , the transition will be determined by an operator that has (move ?dir) and (isconnected ?from ?to ?dir), (agentat ?from) as preconditions, for example.
now, if we want to use an off-the-shelf PDDL planner like fastforward with our modified PDDL files, we need to make sure that it is possible to satisfy the preconditions of the operators, and in particular, the possible high-level actions like (move north) should appear in the state. so if we add all ground action literals to the initial state, we will get the same planning behavior that we would have originally. that’s why we have action literals in the PDDL files